integration-tests/src/test/java/oracle/weblogic/kubernetes/ItHorizontalPodAutoscalerCustomMetrics.java

```diff
@@ -0,0 +1,391 @@
+@IntegrationTest
+@Tag("oke-parallel")
+@Tag("kind-parallel")
+public class ItHorizontalPodAutoscalerCustomMetrics {
+  private static String domainNamespace = null;
+  static int replicaCount = 2;
+  static String wlClusterName = "cluster-1";
+  static String clusterResName = "hpacustomcluster";
+
+  private static String adminSecretName;
+  private static String encryptionSecretName;
+  private static final String domainUid = "hpacustomdomain";
+  private static String adminServerPodName = String.format("%s-%s", domainUid, ADMIN_SERVER_NAME_BASE);
+  private static String managedServerPrefix = String.format("%s-%s-%s",
+      domainUid, wlClusterName, MANAGED_SERVER_NAME_BASE);
+  static DomainResource domain = null;
+
+  private static String opServiceAccount = null;
+  private static String opNamespace = null;
+
+  private static LoggingFacade logger = null;
+  private static  String monitoringExporterDir;
+  private static  String monitoringExporterSrcDir;
+  private static  String monitoringExporterAppDir;
+  private static NginxParams nginxHelmParams = null;
+  private static int nodeportshttp = 0;
+  private static String nginxNamespace = null;
+  private static final String MONEXP_MODEL_FILE = "model.monexp.yaml";
+  private static final String MONEXP_IMAGE_NAME = "monexp-image";
+  private static final String SESSMIGR_APP_NAME = "sessmigr-app";
+  private static String monitoringNS = null;
+  static PrometheusParams promHelmParams = null;
+  private static String releaseSuffix = "hpatest";
+  private static String prometheusReleaseName = "prometheus" + releaseSuffix;
+  private static String prometheusAdapterReleaseName = "prometheus-adapter" + releaseSuffix;
+  private static String hostPortPrometheus = null;
+  private static String prometheusDomainRegexValue = null;
+  private static int nodeportPrometheus;
+  private Path targetHPAFile;
+  private HelmParams prometheusAdapterHelmParams = null;
+
+  /**
+   * Assigns unique namespaces for operator and domains.
+   * Pull WebLogic image if running tests in Kind cluster.
+   * Installs operator.
+   *
+   * @param namespaces injected by JUnit
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(4) List<String> namespaces) {
+    logger = getLogger();
+    monitoringExporterDir = Paths.get(RESULTS_ROOT,
+        "ItMonitoringExporterWebApp", "monitoringexp").toString();
+    monitoringExporterSrcDir = Paths.get(monitoringExporterDir, "srcdir").toString();
+    monitoringExporterAppDir = Paths.get(monitoringExporterDir, "apps").toString();
+    logger.info("Assign a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Assign a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace is null");
+    domainNamespace = namespaces.get(1);
+    logger.info("Get a unique namespace for monitoring");
```

Comment:
Get a unique namespace for monitoring --> Get a unique namespace for monitoring Exporter

Reply:
done